File: 351003/Zhigimont/distcomp/src/main/java/com/bsuir/distcomp/dto/CommentRequestTo.java
package com.bsuir.distcomp.dto;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Data;

@Data
public class CommentRequestTo {

    private Long id;

    @NotNull
    private Long topicId;

    @Size(min = 2, max = 2048)
    private String content;

}

File: 351003/Soika/entities-app/discussion/src/main/kotlin/com/example/entitiesapp/dto/CommentRequestTo.kt
package com.example.entitiesapp.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Size
import java.io.Serializable

data class CommentRequestTo(
    val id: Long? = null,

    @field:NotBlank
    @field:Size(min = 2, max = 2048)
    val content: String,

    @field:NotNull
    val storyId: Long,

    val country: String? = null,
    val state: CommentState = CommentState.PENDING
) : Serializable